### ISSUE TITLE
Javamelody 1.72 (spring boot starter) and Spring Security @Preauthorize FAILS -> It provokes ERROR 404 and it doesn't work

### ISSUE THREAD

**Issue by gonzalopinto** [javamelody/javamelody#756]
Hi, 

I have an application that uses Spring Boot (latest versions) , Spring Security and Javamelody 1.72 (Spring boot starter version).
If I use this version, Javamelody works perfectly but every RestController with @Preauthorize or @Secured start to fail -> ERROR 404 for ALL these methods.
If I use the regular Javamelody version (not spring boot starter) and the JavaMelody Configuration class in order to modify "system-actions-enabled" to false, it happens the same.

The only way to work with everything working fine is to use the regular version but without any configuration class, and I need to modify the system-actions-enabled parameter by CODE, not server configuration, not web.xml (etc), only by CODE, it doesn't work.
If I comment all these @preauthorize methods, then it works perfectly but this is not the right way.

I have search all over the internet and nobody seems to have a proper answer, please help!

Thanks a lot in advance and best regards.

**Comment by evernat** [comment]
Can you give the source of a simple project to reproduce the issue?

**Comment by gonzalopinto** [comment]
It is not easy to reproduce the issue in a simple project, or at least I don't know how to do it.
I have tried to run my project removing all the **@Preauthorized** entries in the controllers and doing this the project works perfectly along the javamelody (spring boot library).

But when I put just one **@Preauthorized** in a single method of the RestController, all the project start to fail again with the famous 404 error.
Here is an example of the situation from my code:

**WORKING:**

	//@PreAuthorize("hasAuthority('APPR_ANACREDIT_SERV_COORD')")
	@RequestMapping(value = "/admin/stepdetails", method = RequestMethod.POST)
	public ResponseEntity<String> findStepDetailsByMonitoringItemDetail(
			@RequestBody final MonitoringItemDetail monitoringItemDetail)
	{
		try
		{
			String stepDetails = this.batchMonitoringServiceAdapter
					.findStepDetailsByMonitoringItemDetail(monitoringItemDetail);
			if (stepDetails == null || stepDetails.isEmpty())
			{
				return new ResponseEntity<>(HttpStatus.NOT_FOUND);// You may decide to return HttpStatus.NO_CONTENT
			}
			return new ResponseEntity<>(stepDetails, HttpStatus.OK);
		}
		catch (BatchMonitoringServiceException e)
		{
			ACRLogger.logError("Error while finding step details (load). Step execution ID: "
					+ monitoringItemDetail.getStepExecutionId(), e);
			return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
		}
	}

**NOT WORKING**

	@PreAuthorize("hasAuthority('APPR_ANACREDIT_SERV_COORD')")
	@RequestMapping(value = "/admin/stepdetails", method = RequestMethod.POST)
	public ResponseEntity<String> findStepDetailsByMonitoringItemDetail(
			@RequestBody final MonitoringItemDetail monitoringItemDetail)
	{
		try
		{
			String stepDetails = this.batchMonitoringServiceAdapter
					.findStepDetailsByMonitoringItemDetail(monitoringItemDetail);
			if (stepDetails == null || stepDetails.isEmpty())
			{
				return new ResponseEntity<>(HttpStatus.NOT_FOUND);// You may decide to return HttpStatus.NO_CONTENT
			}
			return new ResponseEntity<>(stepDetails, HttpStatus.OK);
		}
		catch (BatchMonitoringServiceException e)
		{
			ACRLogger.logError("Error while finding step details (load). Step execution ID: "
					+ monitoringItemDetail.getStepExecutionId(), e);
			return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
		}
	}

Any clue seeing this? Sorry for giving only this example...

Thanks in advance and best regards!

**Comment by evernat** [comment]
The issue is caused by the auto-proxy.
It is fixed by 21d904d and ready for the next release (1.76).
The fix adds spring-boot-starter-aop as a dependency and sets the `javamelody.advisor-auto-proxy-creator-enabled` property to false.

While waiting for the release, you can workaround the issue:
1. add the following in your application.yml:
 ```yml
 javamelody:
   advisor-auto-proxy-creator-enabled: false
 ```
 2. add a dependency on aop in your pom.xml:
 ```xml
 <dependency>
 	<groupId>org.springframework.boot</groupId>
 	<artifactId>spring-boot-starter-aop</artifactId>
 </dependency>
 ```
